Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayEmulatorProcessTests.cs

```diff
@@ -15,171 +18,180 @@
 using Spectre.Console.Cli;
 using Xunit;
 using Xunit.Abstractions;
+using Amazon.Lambda.TestTool.Tests.Common.Retries;
 
 namespace Amazon.Lambda.TestTool.IntegrationTests;
 
-public class ApiGatewayEmulatorProcessTests : IAsyncDisposable
+public class ApiGatewayEmulatorProcessTests(ITestOutputHelper testOutputHelper) : IAsyncLifetime
 {
-    private readonly Mock<IEnvironmentManager> _mockEnvironmentManager = new Mock<IEnvironmentManager>();
-    private readonly Mock<IToolInteractiveService> _mockInteractiveService = new Mock<IToolInteractiveService>();
-    private readonly Mock<IRemainingArguments> _mockRemainingArgs = new Mock<IRemainingArguments>();
-    private readonly ITestOutputHelper _testOutputHelper;
-    private readonly ConcurrentQueue<string> _logMessages = new ConcurrentQueue<string>();
-    private readonly SemaphoreSlim _logSemaphore = new SemaphoreSlim(1, 1);
-    private Process? _lambdaProcess;
-
-    public ApiGatewayEmulatorProcessTests(ITestOutputHelper testOutputHelper)
-    {
-        _testOutputHelper = testOutputHelper;
-    }
+    private readonly Mock<IEnvironmentManager> _mockEnvironmentManager = new();
+    private readonly Mock<IToolInteractiveService> _mockInteractiveService = new();
+    private readonly Mock<IRemainingArguments> _mockRemainingArgs = new();
+    private Task<int>? _emulatorTask;
+    private CancellationTokenSource _cancellationTokenSource = new();
 
-#if DEBUG
-    [Fact]
-#else
-    [Fact(Skip = "Skipping this test as it is not working properly.")]
-#endif
+    [RetryFact]
     public async Task TestLambdaToUpperV2()
     {
         var (lambdaPort, apiGatewayPort) = await GetFreePorts();
-
-        var testProjectDir = Path.GetFullPath("../../../../../testapps");
-        var config = new TestConfig
-        {
-            TestToolPath = Path.GetFullPath(Path.Combine(testProjectDir, "../src/Amazon.Lambda.TestTool")),
-            LambdaPath = Path.GetFullPath(Path.Combine(testProjectDir, "LambdaTestFunctionV2")),
-            FunctionName = "LambdaTestFunctionV2",
-            RouteName = "testfunction",
-            HttpMethod = "Post"
-        };
-
-        var cancellationTokenSource = new CancellationTokenSource();
-
+        _cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(30));
+        var consoleError = Console.Error;
         try
         {
-            StartTestToolProcess(ApiGatewayEmulatorMode.HttpV2, config, lambdaPort, apiGatewayPort, cancellationTokenSource);
+            Console.SetError(TextWriter.Null);
+            await StartTestToolProcessAsync(ApiGatewayEmulatorMode.HttpV2, "testfunction", lambdaPort, apiGatewayPort, _cancellationTokenSource);
             await WaitForGatewayHealthCheck(apiGatewayPort);
-            await StartLambdaProcess(config, lambdaPort);
+            var handler = (APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context) =>
```

Comment:
shouldnt we delete the whole testapps folder then if we are adding all of the handlers in here?

Reply:
I deleted them